### ISSUE TITLE
After successful docker rmi, docker images still reports the image

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#5231]
Test 11-01 from: https://ci.vcna.io/vmware/vic/10830

[integration_logs_10830_d95d6646472f25402a39ae3f74a0209c8ae80512.zip](https://github.com/vmware/vic/files/1025980/integration_logs_10830_d95d6646472f25402a39ae3f74a0209c8ae80512.zip)

**Comment by mhagen-vmware** [comment]
@mhagen looks like rmi untagged the harbor-busybox:latest image but didn't delete the layers because the docker hub busybox is the same image id

But I'm not sure if that's the correct behavior because it's still showing that image with a <none> tag. I'll dig deeper in that piece of code (edited)

**Comment by fdawg4l** [comment]
Vanilla `docker`
```
$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
golang              1.8                 b2063493320d        20 hours ago        698.6 MB
golang              latest              b2063493320d        20 hours ago        698.6 MB

$ docker rmi golang:1.8
Untagged: golang:1.8

$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
golang              latest              b2063493320d        20 hours ago        698.6 MB

$ docker rmi golang
Untagged: golang:latest
Deleted: sha256:b2063493320dce4d554a026d15a16130ad0d2657a30788e980312848048a3eff
Deleted: sha256:488e2d7375e8f2158974a7c2dccfaa5cdc9cf415b6e550ba2d9e09614c5437a8
Deleted: sha256:8bd400ce156213a43920ae9271384b1dc928785d574ece002d670e0aa8263758
Deleted: sha256:7e310c1f1653dc0aa453549be7e3dc8b76dfe80e0eebb7468f438428e5895827
Deleted: sha256:3a78069641bd29f3ad0f11282d93ccbad79de08e280765190f86c33dee1051a5
Deleted: sha256:b625071c6c1618e396e2985a37c0302917c5d1004ee276c032a03642c99c5c8d
Deleted: sha256:10fe56cc338ddcf34cf8e29efd0512d076b3e5f1df99775a230c78b49a4eae68
Deleted: sha256:8d4d1ab5ff74fc361fb74212fff3b6dc1e6c16d1e1f0e8b44f9a9112b00b564f

$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
```

VIC

```
$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
golang              1.8                 b2063493320d        20 hours ago        698.6 MB
golang              latest              b2063493320d        20 hours ago        698.6 MB
golang              <none>              b2063493320d        20 hours ago        698.6 MB

$ docker rmi golang:latest
Untagged: golang:latest

$ docker images                                                                                                                                                                                                                                               
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
golang              1.8                 b2063493320d        20 hours ago        698.6 MB
golang              <none>              b2063493320d        20 hours ago        698.6 MB

$ docker rmi golang:1.8
Deleted: 421938132e69411cdefb982f221b091787ad344fe01d23475935aa677a330ac4
Deleted: a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
Deleted: 722b17304476dd5fc2d68caafa63231d098cd0ef47e7f1a39c62bcfa6c70ca7d
Deleted: a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
Deleted: a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
Deleted: 86e85c214a6d14204fa0505efc3127bf70e7a48e941f6b0b774c7cb83c954b0e
Deleted: a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
Deleted: 96109dbc0c87efa909696eecf0f13ae03eb8c777638a5593f49652f718ae27a1
Deleted: 9021b2326a1e3a942223c7e349a92203df184f2dcca45f5be7b0b80ac50e2ccf
Deleted: fb5937da9414eeab6d68ce06a7ff60d8be1e2c1518ac2588d5df135ab54a9801
Deleted: a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
Deleted: 10a267c67f423630f3afe5e04bbbc93d578861ddcc54283526222f3ad5e895b9
Untagged: golang:1.8

$ docker images                                                                                                                                                                                                                                               
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE

**Comment by cgtexmex** [comment]
@fdawg4l what version of the client were you using?

**Comment by cgtexmex** [comment]
The issue has to do with the fact that were not removing the digest for this `repo:tag` from our image repository.  We only remove the tag where as vanilla docker removes both as demonstrated here:

```
docker rmi 192.168.251.132:5000/jojo:1
Untagged: 192.168.251.132:5000/jojo:1
Untagged: 192.168.251.132:5000/jojo@sha256:c79345819a6882c31b41bc771d9a94fc52872fa651b36771fbe0c8461d7ee558
```

**Comment by cgtexmex** [comment]
Potentially fixed by https://github.com/vmware/vic/pull/5194 -- running a `full ci` to confirm.

**Comment by anchal-agrawal** [comment]
Looks like this is indeed fixed by #5194 - I'll close this issue after that PR is merged.
